crypto/hash/htr/hashtree.go

```diff
@@ -0,0 +1,19 @@
+package htr
+
+import (
+	"github.com/prysmaticlabs/gohashtree"
+)
+
+// VectorizedSha256 takes a list of roots and hashes them using CPU
+// specific vector instructions. Depending on host machine's specific
+// hardware configuration, using this routine can lead to a significant
+// performance improvement compared to the default method of hashing
+// lists.
+func VectorizedSha256(arrayList [][32]byte) [][32]byte {
```

Comment:
I think it's easier to have the caller deciding the destination slice, we can use the same signature as the hasher (or even the hasher itself and avoid the double jump here) and pass the trie layer entirely: 
```go
func VectorizedSha256(outputLayer [][32]byte, inputChunks[][32]byte) {}
```

Reply:
Good point, let me see if it can work better that way.